pages/docs/reference/compatibility-guide-14.md

```diff
@@ -258,6 +258,109 @@ Remember that those definitions are given only for pure Kotlin. Compatibility of
 > `-XXLanguage:-NewInference` can be used to temporarily revert to pre-1.4 behavior. Note that this flag will also
 > disable several new language features.
 
+### Preserve intersection type for covariant types after unchecked cast
+ 
+> **Issue**: [KT-37280](https://youtrack.jetbrains.com/issue/KT-37280)
+> 
+> **Component**: Core language
+> 
+> **Incompatible change type**: source
+> 
+> **Short summary**: since Kotlin 1.4, uchecked casts of covariant types produce the intersection type, not the type used in the cast. 
```

Comment:
Let's add something like "produce the intersection type for smart casts", that last part about smart casts is important